lib/raxx/context/server_context.ex

```diff
@@ -0,0 +1,53 @@
+defmodule Raxx.Context.ServerContext do
+  @typedoc """
+  `"http1.1" | "http2"`
+  """
+  @type http_version :: String.t()
+  @type transport :: :tls | :tcp
+  @type ip_address :: :inet.ip4_address() | :inet.ip6_address()
+  @type port_number :: :inet.port_number()
+
+  @moduledoc """
+  The server context is a place where the HTTP Server implementing the
+  Raxx interface can put arbitrary information that doesn't fit elsewhere.
+  That could be anything from low-level network information to
+  server-specific debug information.
+
+  ## Standard fields
+
+  There's a number of standard properties that are common between HTTP servers.
+  If a server is to put any of them in the server context, it should follow
+  the standard format:
+
+  | *map key* | *type* | *description* |
+  | `:remote_ip_address` | `t:ip_address/0` | the ip address of the remote machine |
+  | `:local_port_number` | `t:port_number/0` | the local port number where the request was sent |
```

Comment:
I think `remote_port_number` is a thing

Reply:
It is for any TCP connection, but I don't think it would be relevant or actionable in this context. Even if the user knows the remote port, I don't think there's anything useful they could do with it, as in general it would be a randomised port that nothing would be listening on. https://stackoverflow.com/a/29045432

Of course there would be nothing wrong with a HTTP server exposing this information, but I wouldn't go as far as talking about it in the docs here.